app/code/community/IyzicoCheckoutForm/IyzicoCheckoutForm/Model/Cancel/Observer.php

```diff
@@ -0,0 +1,93 @@
+
+    private $_configuration;
+    private $_request;
+    private $_conversationId = '';
+    private $_paymentId = '';
+
+    public function iyzicoOrderCancel(Varien_Event_Observer $observer) {
+        $paymentMethod = $observer->getPayment()->getMethodInstance()->getCode();
+        if ($paymentMethod == IyzicoCheckoutForm_IyzicoCheckoutForm_Helper_Data::IYZICO_CREDITCARD) {
+            $currentCurrencyCode = Mage::app()->getStore()->getCurrentCurrencyCode();
+            $currencyArr = Mage::helper('iyzicocheckoutform')->getCurrencyArr();
+            if (!in_array($currentCurrencyCode, $currencyArr)) {
+                Mage::throwException(Mage::helper('iyzicocheckoutform')->__("Please make sure that the currency value in your payment request is one of the (USD, EUR, GBP, IRR, TL) valid values"));
+            }
+
+            $customInfo = $observer->getPayment()->getAdditionalInformation('custom_info');
+            $customInfoArr = unserialize($customInfo);
+            $this->_conversationId = $customInfoArr['conversation_id'];
+            $this->_paymentId = $customInfoArr['payment_id'];
+
+            $payment = $observer->getEvent()->getPayment();
+            $order = $payment->getOrder();
+            $orderData = $order->getOrigData();
+            $orderIncrementId = $orderData['increment_id'];
+
+            if (($order->getStatus() != 'pending_payment') && (!empty($this->_paymentId))) {
+                $this->_initIyzipayBootstrap();
+                $this->_setClientConfiguration();
+                $this->_setCreateRequest();
+				
+                $apiLogData = array(
+                    'method_type' => IyzicoCheckoutForm_IyzicoCheckoutForm_Helper_Data::CANCEL_API,
+                    'order_increment_id' => $orderIncrementId,
+                    'request_data' => $this->_request->toJsonString(),
+                    'response_data' => '',
+                    'status' => 'pending',
+                    'created' => date('Y-m-d H:i:s'),
+                    'modified' => date('Y-m-d H:i:s'),
+                );
+
+                $lastInsertedId = Mage::helper('iyzicocheckoutform')->saveIyziTransactionApiLog($apiLogData);
+
+                $response = \Iyzipay\Model\Cancel::create($this->_request, $this->_configuration);
+
+                Mage::helper('iyzicocheckoutform')->saveIyziTransactionApiLog(array('response_data' => $response->getRawResult(), 'status' => $response->getStatus()), $lastInsertedId);
+                $cancelPaymentStatus = $response->getStatus();
+                if ('success' != $cancelPaymentStatus) {
+                    $orderId = $order->getEntityId();
+                    $order->setActionFlag(Mage_Sales_Model_Order::ACTION_FLAG_CANCEL, false);
+                    Mage::getSingleton('core/session')->getMessages(true);
+                    Mage::getSingleton('core/session')->addError('There are some internal error, please try again.');
```

Comment:
Burdaki hata olmayacak hatalara verişmiş ve geçiştirilmiş yani seninde anladığın gibi iyzico dan eğer işlem başarısız ise error mesajı dönmeme ihtimali yok ama yinede bir kontrol yaptırılmış mesela burda ödeme alındımı  durumunun successmi diye sorgulamış böyle bir hata olmasına imkan yok istersen bunların hepsi için bana bir türkçe bir mesaj gönder hepsine aynısını yazıyım ben bunları gördüm fakat güncellemedim önemsiz diye